### ISSUE TITLE
Is CV working?

### ISSUE THREAD

**Issue by namankumar** [automl/auto-sklearn#196]
Looking at automl.grid_scores_, I am expecting 1 value for mean_validation_score and >5 values for cv_validation_scores. But what I see is 1 value for mean_validation_score and 1 value (same value) for cv_validation score. So, I'm not sure if CV is working?  See my constructor below

Related questions: 
CV is used by auto-sklearn to validate hyperparameters, correct?
After getting predictions, I should determine model generalization with my testing set? Do you recommend any particular strategy? My current method for auto-sklearn is: 1. init constructor 2. fit 3. refit 4. fit with testing data 5. determine generalization by selective sampling and iterations.

Here is the constructor:
```

automl = autosklearn.regression.AutoSklearnRegressor(
         time_left_for_this_task=5400, per_run_time_limit=1000,
         tmp_folder='/Users/naman/automl/xfiles/temp,
         output_folder='/Users/naman/automl/xfiles/out,
         ml_memory_limit = 1000,
         delete_tmp_folder_after_terminate=False,
         delete_output_folder_after_terminate=False,
         resampling_strategy='cv',
         resampling_strategy_arguments={'folds': 5},
         shared_mode=True)


```

**Comment by mfeurer** [comment]
* CV is working, but the fold information is not propagated to the `grid_scores_` attribute. 
* Yes, CV is used to validate hyperparatemeters (get an unbiased estimate of a hyperparameter configuration).
* I don't understand step four and five of your workflow. We're usually first doing a fit with CV on the training data set and then do a refit on the whole training data set. We use this refitted classifier to predict for the test set, which gives us the generalization error.

**Comment by namankumar** [comment]
Got it, thanks! Closing this.

In step 4, I predict with testing data and get the r2 score

In step 5, I do 5 fold CV with all of my data. Step 5 is probably unconventional, I have a very small dataset (100 samples) so I do step 5 to test generalization across different views of the data.
